File: 29-04/exponenciacao.py
def exponenciacao(a,b):
    if b == 0:
        return 1
    elif b > 0:
        return a * exponenciacao(a, b-1)

if __name__ == "__main__":
    print("Digite a base:")
    a = int(input())
    print("Digite o expoente:")
    b = int(input())
    print("A exponenciacao é:")
    if b < 0:
        print("1 /", exponenciacao(a, -b))
    else:
        print(exponenciacao(a, b))

File: 29-04/exponenciacao.hs
exponenciacao a b
    | b == 0 = 1
    | b > 0 = a * (exponenciacao a (b - 1))

main :: IO () 
main = do
    putStrLn "Digite a base:"
    base <- readLn
    putStrLn "Digite o expoente:"
    expoente <- readLn
    putStrLn "A exponenciacao é:"
    if expoente < 0 then
        putStrLn (show 1 ++ "/" ++ show (exponenciacao base (-expoente)))
    else
        print (exponenciacao base expoente)

